### ISSUE TITLE
[Bug]: Unable to use applyRowMutations() in bigquery IO apache beam java

### ISSUE THREAD

**Issue by vs0055** [apache/beam#27616]
### What happened?

Create two tables in bigquery with primary key on table1 and table2 then create simple pipeline to read from table1 and upsert into table 2 with same schema 

The primary key is will be a string with random unique value

```
PCollection<TableRow> rows1 = p.apply(BigQueryIO.readTableRows().from(" project.dataset.table1"));


PCollection<RowMutation> rowmutate = rows1.apply(MapElements
                .into(new TypeDescriptor<RowMutation>() {
                })
                .via(tableRow -> RowMutation.of(tableRow,
                        RowMutationInformation.of(RowMutationInformation.MutationType.UPSERT, new Random().nextLong(1,1000000)))));


        rowmutate.setCoder(RowMutation.RowMutationCoder.of())

                .apply(BigQueryIO.applyRowMutations()
                        .to(" project.dataset.table2")
                        .withSchema(createSchema())
                        .withMethod(BigQueryIO.Write.Method.STORAGE_API_AT_LEAST_ONCE)
                .withCreateDisposition(BigQueryIO.Write.CreateDisposition.CREATE_NEVER))
```
I am facing these two exceptions 

`com.google.api.gax.rpc.InvalidArgumentException: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: The primary keys and clustering keys of the table are required to create an upsert stream. Entity: projects/myproject/datasets/mydataset/tables/mytable/streams/_default`

`Caused by: java.lang.RuntimeException: Append to stream projects/myproject/datasets/dataset/tables/table/streams/_default failed with stream doesn't exist
`

### Issue Priority

Priority: 1 (data loss / total loss of function)

### Issue Components

- [ ] Component: Python SDK
- [X] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [X] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

**Comment by liferoad** [comment]
@Abacn @ahmedabu98 Can you check this?

**Comment by Virinchi2595** [comment]
@reuvenlax need your assistance here can you please check what is issue

**Comment by reuvenlax** [comment]
have you set a primary key on this table when creating it?

**Comment by Virinchi2595** [comment]
> have you set a primary key on this table when creating it?

yes @reuvenlax

**Comment by liferoad** [comment]
post his stack overflow question here: https://stackoverflow.com/questions/76739967/apache-beam-bigqueryiojavaio-grpc-statusruntimeexceptioninvalid-argumentthe?noredirect=1#comment135312528_76739967

**Comment by reuvenlax** [comment]
The tablespec in to should be in the form <project>.<dataset>.<table>

**Comment by Virinchi2595** [comment]
> The tablespec in to should be in the form project.dataset.table

Yes I used table spec in same format(project.dataset.table) in the issue i mentioned table1 and table2 just for reference.

**Comment by kota-yata** [comment]
The project id and the dataset name should be joined together with colon, not with period. `myproject:testing.test_new` instead of `myproject.testing.test_new`.

The code above should be as follows:
```
rows.apply(BigQueryIO.applyRowMutations()
                        .to("myproject:testing.test_new")
                        .withJsonSchema(tableSchemaJson)
                        .withMethod(BigQueryIO.Write.Method.STORAGE_API_AT_LEAST_ONCE)
                .withCreateDisposition(BigQueryIO.Write.CreateDisposition.CREATE_NEVER))
```

I've actually posted my answer to the stack overflow question.

**Comment by colt-snaphade** [comment]
Hello guys, anyone got solution on this.  I'm also facing the exact same issue.
